src/Performance/Minibench.purs

```diff
@@ -35,6 +40,45 @@ withUnits t
   | t < 1.0e9 = toFixed (t / 1.0e6) <> " ms"
   | otherwise = toFixed (t / 1.0e9) <> " s"
 
+-- | When running `bench` or `benchAff`, this is the number of samples that will
+-- | be used.
+defaultNumSamples :: Int
+defaultNumSamples = 1000
+
+-- | The `Results` type is a `Record` newtype that contains information about
+-- | the timings for the test run.
+newtype Results = Results
+  { mean :: Number
+  , stddev :: Number
+  , min' :: Number
+  , max' :: Number
+  }
+
+instance showResults :: Show Results where
+  show (Results results') =
+    "mean   = " <> withUnits results'.mean <> "\n" <>
+    "stddev = " <> withUnits results'.stddev <> "\n" <>
+    "min    = " <> withUnits results'.min' <> "\n" <>
+    "max    = " <> withUnits results'.max'
+
+-- | Given an array of run timings, calculate the results for the test.
+results :: Array Number -> Results
+results runs = Results {mean, stddev, min', max'}
+  where
+    sum = foldr (+) 0.0 runs
+    sum2 = foldr (\cur -> (+) (cur * cur)) 0.0 runs
+    n = toNumber $ length runs
+    mean = sum / n
+    stddev = sqrt ((sum2 - n * mean * mean) / (n - 1.0))
+    min' = foldr min infinity runs
+    max' = foldr max 0.0 runs
+
+-- | Run a synchronous function n times, returning the array of runtimes.
+runSync :: forall a. Int -> (Unit -> a) -> Array Number
+runSync n f = flip map (0..n) \_ -> fromHrTime $ runPure do
```

Comment:
Perhaps we ought to not use `Eff` for `hrTime` and just make it into a pure function, since we're treating it like one anyway.

Reply:
Since measuring hrtime is impure, I think it would be a lie to make it a pure function.  I'd prefer to add an effect type for it and use `unsafePerformEff`.

Either way, I think that enough is changing for one PR, so I'd rather do whatever we decide on in a follow-up PR instead of wrapping it into this one.  Thoughts?